samples/bot-all-cards/csharp/BotAllCards/Dialogs/AllCards.cs

```diff
@@ -0,0 +1,157 @@
+
+            var thumbnailCardAttachment = new Attachment()
+            {
+                ContentType = "application/vnd.microsoft.card.thumbnail",
+                Content = JsonConvert.DeserializeObject(thumbnailCardJson),
+            };
+
+            return thumbnailCardAttachment;
+        }
+
+        /// <summary>
+        /// This card enables a bot to request that a user signs in.
+        /// </summary>
+        /// <returns>Return Microsoft.Bot.Schema.SigninCard results.</returns>
+        public static SigninCard GetSigninCard()
+        {
+            var signinCard = new SigninCard
+            {
+                Text = "BotFramework Sign-in Card",
+                Buttons = new List<CardAction>
+                {
+                    new CardAction(
+                        ActionTypes.OpenUrl, "Sign-in", value: "https://login.microsoftonline.com/") },
+            };
+
+            return signinCard;
+        }
+
+        /// <summary>
+        /// This card collection is used to return multiple items in a single response.
+        /// </summary>
+        /// <returns>Return Microsoft.Bot.Schema.Attachment results</returns>
+        public static Attachment CollectionsCardAttachment()
+        {
+            var paths = new[] { ".", "Resources", "collectionsCard.json" };
+            var collectionsCardJson = File.ReadAllText(Path.Combine(paths));
+
+            var collectionsCardAttachment = new Attachment()
+            {
+                ContentType = "application/vnd.microsoft.card.adaptive",
+                Content = JsonConvert.DeserializeObject(collectionsCardJson),
+            };
+
+            return collectionsCardAttachment;
+        }
+
+        /// <summary>
+        /// This card has a flexible layout with multiple sections, fields, images, and actions.
+        /// </summary>
+        /// <returns>Return Microsoft.Bot.Schema.Attachment results</returns>
+        public static Attachment Office365ConnectorCard()
+        {
+            var paths = new[] { ".", "Resources", "o365ConnectorCard.json" };
+            var OfficeCardJson = File.ReadAllText(Path.Combine(paths));
+
+            var OfficeCardAttachment = new Attachment()
+            {
+                ContentType = "application/vnd.microsoft.teams.card.o365connector",
+                Content = JsonConvert.DeserializeObject(OfficeCardJson),
+            };
+
+            return OfficeCardAttachment;
+        }
+
+        /// <summary>
+        /// This card enables a bot to request that a user OAuth Card
+        /// </summary>
+        /// <returns>>Return Microsoft.Bot.Schema.Attachment results</returns>
+        public static OAuthCard GetOAuthCard()
+        {
+            var oauthCard = new OAuthCard
+            {
+                Text = "BotFramework OAuth Card",
+                ConnectionName = "YOUR-CONNECTION-NAME", // Replace with the name of your Azure AD connection.
```

Comment:
Read it from the appsettings only otherwise auto deployment will fail

Reply:
Fixed